giving-example/public/adyenGiving.js

```diff
@@ -0,0 +1,86 @@
+const clientKey = document.getElementById("clientKey").innerHTML;
+
+async function callServer(url, data) {
+  const res = await fetch(url, {
+    method: "POST",
+    body: data ? JSON.stringify(data) : "",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  return await res.json();
+}
+
+async function handleDonation(donationToken, pspReference, amount) {
+  try {
+    const res = await callServer(`/api/donations?donationToken=${encodeURIComponent(donationToken)}&pspReference=${pspReference}`, amount);
```

Comment:
From security point-of-view, is there any reason why we don't add both the donationToken and the pspReference (==originalPspReference from the payment) in the POST payload (similar to amount). Now this is exposed in the URL.

Should the donationToken be kept a secret?

Reply:
Refactored to store donation token in session

Reply:
Nice work